.github/workflows/continuous-deploy.yml

```diff
@@ -332,32 +332,30 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      # - name: Determine branch
-      #   id: set_branch
-      #   uses: actions/github-script@v5
-      #   with:
-      #     script: |
-      #       const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
-      #       const octokit = require('@actions/github').getOctokit(GITHUB_TOKEN);
-      #       const branchName = process.env.GITHUB_REF.replace('refs/heads/', '');
-      #       try {
-      #         await octokit.rest.repos.getBranch({
-      #           owner: process.env.GITHUB_REPOSITORY_OWNER,
-      #           repo: (process.env.GITHUB_REPOSITORY).split('/')[1],
-      #           branch: branchName,
-      #         });
-      #         return branchName;
-      #       } catch (error) {
-      #         return 'master';
-      #       }
+      - name: Authenticate GitHub CLI
+        run: echo "${{ secrets.PAT }}" | gh auth login --with-token
+
+      - name: Check if branch exists
+        id: check_branch
+        env:
+          BRANCH_NAME: ${{ github.head_ref }}
+          DEFAULT_BRANCH: master
+        run: |
+          if gh api repos/didx-xyz/charts/git/ref/heads/$BRANCH_NAME; then
+            echo "::set-output name=branch_exists::true"
+            echo "::set-output name=branch_name::$BRANCH_NAME"
```

Comment:
[GitHub Actions: Deprecating save-state and set-output commands](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)